packages/wallet/src/TxBuilder/buildTx.ts

```diff
@@ -0,0 +1,171 @@
+import { Cardano, NotImplementedError } from '@cardano-sdk/core';
+import { firstValueFrom } from 'rxjs';
+
+import {
+  DeepPartial,
+  MaybeValidTx,
+  OutputBuilder,
+  OutputValidationMinimumCoin,
+  OutputValidationTokenBundleSize,
+  SignedTx,
+  TxBodyValidationError,
+  TxBuilder,
+  TxOutValidationError
+} from './types';
+import { InitializeTxResult, ObservableWallet } from '../types';
+import { OutputBuilderSaw, toOutputValidationError } from './OutputBuilder';
+import { OutputValidator, StakeKeyStatus, createWalletUtil, deepEquals } from '../services';
+import { SignTransactionOptions } from '../KeyManagement';
+
+/**
+ * Minimal sub-type of ObservableWallet that is used by TxBuilder
+ */
+export type TxBuilderObservableWallet = Pick<
+  ObservableWallet,
+  'utxo' | 'protocolParameters$' | 'finalizeTx' | 'initializeTx' | 'submitTx' | 'delegation'
```

Comment:
A potential improvement here is to narrow down the types of some of the picked properties too